clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -485,6 +486,8 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
                 String desc = channel.socketDescription();
                 if (e instanceof IOException)
                     log.debug("Connection with {} disconnected", desc, e);
+                else if (e instanceof AuthenticationException) // will be logged later as error by clients
```

Comment:
`Will be logged` or `will be thrown`?

Reply:
It is `logged` and `thrown`. The comment refers to the fact that we don't log it here at a higher level even though it is a critical exception because it is logged at error level later.